diff --git a/mm/mmap.c b/mm/mmap.c
index f875386e7acd..4ed4d4381331 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -1679,6 +1679,9 @@ unsigned long mmap_region(struct file *file, unsigned long addr,
 	struct rb_node **rb_link, *rb_parent;
 	unsigned long charged = 0;
 
+	if (!jiffies)
+		printk("kpatch mmap foo\n");
+
 	/* Check against address space limit. */
 	if (!may_expand_vm(mm, vm_flags, len >> PAGE_SHIFT)) {
 		unsigned long nr_pages;
